source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm

```diff
@@ -122,16 +122,24 @@ - (UIView* )render:(UIView* )viewGroup
      andHostConfig:(std::shared_ptr<HostConfig> const &)config
 {
     std::shared_ptr<Image> imgElem = std::dynamic_pointer_cast<Image>(elem);
+    NSString* urlStr = [NSString stringWithCString: imgElem->GetUrl().c_str()
+                                          encoding:[NSString defaultCStringEncoding]];
+    NSURL* url = nil;
+    NSRange range = [urlStr rangeOfString:@"%"];
+    //check if % is already embedded in urlStr
+    if(range.length > 0)
+    {
+        url = [NSURL URLWithString:urlStr];
+    }
+    else
+    {
+        url = [NSURL URLWithString: 
+                [urlStr stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLFragmentAllowedCharacterSet]];
```

Comment:
>NSCharacterSet.URLFragmentAllowedCharacterSet [](start = 75, length = 45)

This set of characters are the ones that are allowed in the fragment. It's not correct to apply this to the whole URL.

For example, the '=' character is allowed in the fragment but not in query elements. The '#' character is not allowed in the fragment but is allowed elsewhere, as the fragment delimiter.